include/aspect/material_model/grain_size.h

```diff
@@ -201,12 +201,56 @@ namespace aspect
          * set to false we use the approach of Hall, C. E.,
          * Parmentier, E. M. (2003). Influence of grain size evolution on
          * convective instability. Geochemistry, Geophysics, Geosystems, 4(3).
+         * Or: TODO add Mulyukova reference.
          */
-        bool use_paleowattmeter;
         std::vector<double> grain_boundary_energy;
         std::vector<double> boundary_area_change_work_fraction;
         std::vector<double> geometric_constant;
 
+        /**
+          * A struct that contains information about which
+          * formulation of grain size evolution should be used.
+          */
+        struct Formulation
+        {
+          /**
+           * This enum lists available options that
+           * determine the dynamic recrystallization.
```

Comment:
```suggestion
           * determine the equations being used for grain size evolution.
```

Since this changes both grain growth and reduction.